gk/main.c

```diff
@@ -1327,6 +1332,9 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
 				continue;
 			}
 
+			instance->tot_pkts_num_dropped++;
+			instance->tot_pkts_size_dropped += pkt_len;
+
 			/* Drop non-IP and non-ARP packets. */
 			drop_packet(pkt);
```

Comment:
I think the original point of the `drop_packet()` function was to centralize all the work done when a packet is dropped, including updating statistics. I think we can pass `instance` to `drop_packet()` and update these statistics there instead of everywhere `drop_packet()` is called.

Reply:
We could also pass `pkt_len` since we already calculated it.

Reply:
``drop_packet()`` will also be called by functions processing the packets from back interface, however, these packets won't go through the granted/request channel. Do we need to do measurements over these packets as well?

Reply:
I think we probably don't need to measure dropped packets coming from the back interface, so perhaps we need a `drop_pkt_front()` and `drop_pkt_back()` or something. My comment was more based on the observation that we keep repeating the pattern <update drop statistics> and then <call `drop_packet()`> which makes me think all the functionality can be pushed into a function to eliminate the duplication.

Reply:
Let's not measure the packets dropped from the back interface for this version of the measurements, but let's have `drop_pkt_back()` for an eventual change.